sys/ddb/db_main.c

```diff
@@ -191,7 +202,7 @@ X_db_symbol_values(db_symtab_t *symtab, c_db_sym_t sym, const char **namep,
 }
 
 int
-db_fetch_ksymtab(vm_offset_t ksym_start, vm_offset_t ksym_end,
+db_fetch_ksymtab(vm_pointer_t ksym_start, vm_pointer_t ksym_end,
```

Comment:
ksym_end can remain a virtual address

Reply:
While it can, it's a bit odd of an interface to pass around a pointer in one case and a VA in the other, and most of the other similar APIs in the kernel linker and other places were simpler to leave as pairs.

Reply:
The other open question is that ksym_start and ksym_end come from the metadata from the loader, and I think we aren't sure yet what the loader will pass: will it pass cap pointers to the kernel (it probably should?) or it will it pass virtual addresses.

Reply:
I imagine it should pass whatever the native ABI of the loaded kernel is.